docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+Each release will have an edition of the document, and the current edition of the document for the first public release is v2.0.
+
+The goal of this document is to cover the high-level system architecture and design. This document is divided into three major parts: design, implementation, product proposition. 
+
+## Design
+
+### Architecture
+
+![diagram](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ArchitectureDiagram.png)
+
+Figure 1. Architecture Diagram
+
+The Architecture Diagram given above explains the high-level design of the App. Given below is a quick overview of each component.
+
+The `Main` class is responsible for,
+* At app launch: Initializes the main components in the correct sequence and read data from our data files
+* At shut down: Terminates the continuous loop and shut down the components
+
+`Common` represents a collection of classes used by multiple other components.
+
+The rest of the App consists of four components.
+* `UserInterface`: The user interface of the App.
+* `Controller`: User input parser and command executor
+* `Model`: Holds the data of the App in memory
+* `Storage`: Reads data from, and writes data to, the hard disk.
+
+Each of the four components:
+The interaction between them is defined in an interface.
+Exposes its functionality using a concrete {Component Name}Manager class.
+
+The sections below give more details of each component.
+
+#### UserInterface component
+
+![userinterfacecomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/UserInterfaceComponent.png)
+
+Figure 2. Class Diagram of the UserInterface component
+
+API: UserInterface.java
+
+The UserInterface component,
+* Prompts commands from the user.
+* Execute user commands using the Controller component.
+* Displays information based on changes to Model data.
+* Prints the user’s events of the week in a timetable format.
+
+This component uses the singleton design, meaning that there is only an instance required throughout the entire lifetime of the application, obtained with the following command
+`UserInterface userInterface = UserInterface.getInstance()`
+<br />
+<br />
+
+#### Controller component
+
+![controllercomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ControllerComponent.png)
+
+Figure 3. Class Diagram of the Controller component
+
+API: ControlManager.java
+
+The Controller component,
+* Receives user input from the user interface.
+* Extracts the command and model type from the user input.
+* Generate the required command from the CommandFactory.java class
+* Extract the required model to be accessed and modified.
+* Executes the command with the corresponding model.
+
+<br />
+<br />
+
+#### Model component
+
+![modelcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/ModelComponent.png)
```

Comment:
![image](https://user-images.githubusercontent.com/60251547/97667528-1183f800-1abb-11eb-9194-5129873b9dcc.png)

can try to re-position the arrows so that the arrows are not covering the words or overlapping?